src/main/java/technology/dice/dicefairlink/driver/AuroraReadReplicasDriver.java

```diff
@@ -273,7 +286,7 @@ private void addDriverForDelegate(String delegate, final String stringURI) throw
 
   private synchronized String getNextReplica(final AuroraReadonlyEndpoint replicaEndPoint) {
     String nextReplica = replicaEndPoint.getNextReplica();
-    if (nextReplica.equals(lastReplica.get())) {
+    if (nextReplica != null && nextReplica.equals(lastReplica.get())) {
```

Comment:
Last replica appears to be a value shared amongst all the clusters the driver is handling.

If there is more than one cluster, then the last one to call `getNextReplica`will compare `lastReplica` with the value returned to another cluster.

In other words. for more than one cluster, this is a bug.

Reply:
The `lastReplica` moved from `AuroraReadReplicasDriver` to `AuroraReadonlyEndpoint` so it will be cluster endpoint specific. No tests for this case, though.